rfcs/ils-0104-library-kpis.md

```diff
@@ -0,0 +1,424 @@
+                 - `available`
+                   - The loan is done via a loan request and the document is available (one available item with `status` `CAN_CIRCULATE`).
+                 - `not_available`
+                   - The loan is done via a loan request but the document is not available (no available item with `status` `CAN_CIRCULATE`).
+               - copy over the fields `loan_creation_method` and `document_availability_during_loan_creation`, so the query can filter on them individually
+                 - e.g. only count loans that were created manually: filter `loan_creation_method=manually`
+
+     2. Average Waiting Time := waiting time between loan (request) creation and loan start time / number of loans started
+          - query: waiting time between loan (request) creation and loan start time
+            - event: loan starts (use signal `invenio_circulation.signals.loan_state_changed`, indicated by the transition to the `state` `ITEM_ON_LOAN` from any other state)
+            - aggregate:
+              - sum of waiting time
+                - waiting time := loan `start_date` - loan (request) creation date
+                  - loan (request) creation date is based on the creation method := 
+                    - `self_checkout` and `manually`: use the field `_created` on the loan
+                    - `interlibrary_loan`: use either the field
+                      - `_created` on the `invenio_app_ils.borrowing_request` that is linked to the loan
+              - monthly
+              - over field `loan_creation_method__document_availability_during_loan_creation`
+                - see 3.1 for definition of `loan_creation_method__document_availability_during_loan_creation`
+                  - we have to extract the value of this field from the events index of KPI 3.1, as the information is only valid during loan creation 
+                - we also add information about the provider to the event for interlibrary loans, so future aggregations can differentiate the waiting time based on the provider
+            - query: number of loans started
+              - event: loan starts (use signal `invenio_circulation.signals.loan_state_changed`, indicated by the transition to the `state` `ITEM_ON_LOAN` from any other state)
+              - aggregate:
+                - count
+                - monthly
+                - over field `loan_creation_method__document_availability_during_loan_creation`
+                  - see 3.1 for definition of `loan_creation_method__document_availability_during_loan_creation`
+                    - we have to extract the value of this field from the events index of KPI 3.1, as the information is only valid during loan creation 
+                  - we also add information about the provider to the event for interlibrary loans, so future aggregations can differentiate the waiting time based on the provider
+
+
+4. Number of changes to the Library collections:
```

Comment:
In this section I believe are missing to store also curators id - we are frequently asked for stats for an individual curator